packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/api/EventGridContext.ts

```diff
@@ -16,11 +16,6 @@ export function createEventGrid(
   credential: KeyCredential,
   options: EventGridClientOptions = {}
 ): EventGridContext {
-  const baseUrl = endpoint;
-  options.credentials = {
-    ...options.credentials,
-    apiKeyHeaderName: "SharedAccessKey",
```

Comment:
This should be expected as we already set SharedAccessKey as default apiKeyHeaderName in the rest client factory here 
https://github.com/Azure/autorest.typescript/blob/main/packages/typespec-test/test/eventgrid_modular/generated/typespec-ts/src/rest/eventGridClient.ts#L24-L26